### PR TITLE
fix(just): harden justfiles and PowerShell bootstrap

### PR DESCRIPTION
**Pull request by thms317**
Hardens the `just` recipes and the Windows PowerShell bootstrap, scoped to **low-risk fixes only** (no refactors).

## justfiles

**Bugs / silent failures**
- **`just lint` could never fail**: the leading `-` on each checker ignored its exit code, so the recipe — and every CI gate that calls it — always passed. Now a `[script]` recipe that runs ruff/ty/pydoclint, reports everything in one pass, and exits non-zero if any fail.
- **`clean` deleted the committed `uv.lock`** (root + template) and re-resolved from scratch. Removed from the prune list; `uv.lock` is a reproducibility artifact.
- **`test-deploy profile=…` was ignored** — the recipe interpolated the `PROFILE_NAME` constant instead of the `{{ profile }}` argument.
- **`run-temporary-deployment-tests`**: `git commit` failed on machines without a git identity (and looked like a hook failure). Added a throwaway `-c user.email/-c user.name`; unsilenced `git add`.

**Diagnostics & docs**
- Dropped `2>&1` on `uv build` (lint/test/validate/deploy) and `2>/dev/null` on the `clean` find so real failures are visible; added `set -e` to the `clean` script recipe.
- Fixed misleading doc-comments: `test` no longer claims to "Run pre-commit hooks"; root `clean` now describes its full behaviour.
- `install-extensions`: added `set -e` and guards for a missing `extensions.json` / empty extension list (would otherwise call `code --force` with no args and "succeed").

## PowerShell (`just_bash.ps1`)
- `Get-GitBashPath` returns `$null` instead of throwing, so the winget install-Git fallback in `EnsureGitBash` is actually reachable.
- Fixed the dead 32-bit probe: `"$env:ProgramFiles(x86)\…"` → `"${env:ProgramFiles(x86)}\…"` (the `(` truncated the variable name, so the path never existed).
- Made the winget upgrade/install path robust across PowerShell hosts (thrown error vs non-zero exit), with the hard version re-check as the real gate.
- Wrapped `git update-index --skip-worktree` in try/catch so a not-yet-tracked checkout doesn't emit a confusing error or abort setup.

## Verification
`ci-dabs` is green on this branch: all six matrix configs exercise `just` / `just setup` / `just lint` / `just test` / `just clean`, and the Windows runners execute `just_bash.ps1` end-to-end. The stricter `just lint` surfaced no latent violations in the generated template.

---
*Two earlier commits were dropped during rebase as superseded: the backslash-import fix landed via #112, and the Node-20 CI action bumps landed via #119.*

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: template/{{.project_name}}/.just/just_bash.ps1 [hunk 1]
```diff
@@ -180,7 +189,15 @@ set script-interpreter := [
...
 
-git update-index --skip-worktree -- ".just/shell_settings.justfile"
+# Hide local edits to the generated shell settings from git. Best-effort: the file may
+# not be tracked yet (e.g. a fresh template init before the first commit), in which case
+# git exits non-zero and writes to stderr. With $ErrorActionPreference = "Stop", the
+# stderr redirect alone can raise a terminating error, so try/catch is required.
+try {
+  git update-index --skip-worktree -- ".just/shell_settings.justfile" 2>$null
+} catch {
+  Write-Output "Skipping skip-worktree: .just/shell_settings.justfile is not tracked yet."
```
Comment (Sander-Brickmans) [inline template/{{.project_name}}/.just/just_bash.ps1]: This is a good point, but if this is true, then in the catch scenario you would want the first commit for a fresh template to be followed by this command, can we include this as well?

File: template/{{.project_name}}/.just/just_bash.ps1 [hunk 2]
```diff
@@ -74,16 +74,24 @@ function EnsureJustMinVersion {
...
+      if ($LASTEXITCODE -ne 0) { $upgradeFailed = $true }
     } catch {
-      $upgradeOk = $false
+      $upgradeFailed = $true
     }
 
-    if (-not $upgradeOk) {
-      winget install --id Casey.Just --source winget --accept-package-agreements --accept-source-agreements
+    if ($upgradeFailed) {
+      Write-Output "winget upgrade did not apply, falling back to install..."
```
Comment (Sander-Brickmans) [inline template/{{.project_name}}/.just/just_bash.ps1:89]: If we are hardening, we may want to add a check after installation to see if we are getting the correct version now. Windows allows multiple commands with the same name to be on Path, and installing another version of the same tool may succeed, while keeping the old version as the one that is called. So you could have `just version 1.56 installed successfully` followed by `just --version = 1.10`. Have seen this happening multiple times, not as much an edge case as you would hope.

File: .justfile [hunk 3]
```diff
@@ -101,7 +102,7 @@ test-deploy profile=PROFILE_NAME:
 		"empower_vscode": "yes" \
 		}' > temporary_deployment/init_params.json;
 	echo "Initializing a new Databricks Asset Bundle from template...";
-	databricks bundle init . --config-file "temporary_deployment/init_params.json" -p {{ PROFILE_NAME }};
+	databricks bundle init . --config-file "temporary_deployment/init_params.json" -p {{ profile }};
```
Comment (Sander-Brickmans) [inline .justfile:107]: why change PROFILE_NAME to profile? Other recipes still seem to use PROFILE_NAME, is there inconsistency somewhere?

The dab.justfile loads the profile from a .env file with a default value,

Comment (Sander-Brickmans) [reply .justfile:107]: I now see this is the repo root justfile, not the template justfile. So the value may very well be different here, but I think it deserves an explicit remark.

**Review by skinbody — DISMISSED** [review]

File: template/{{.project_name}}/.just/just_bash.ps1 (hunk 2 above)
Comment (skinbody) [reply template/{{.project_name}}/.just/just_bash.ps1:89]: This is already covered a few lines below (99-102): after PathRefresh, we call just --version again, parse it, and throw if the resolved command is still below the required version. So if winget installs a newer just but PATH still resolves to the old one, setup still fails rather than silently continuing.

Comment (skinbody) [reply template/{{.project_name}}/.just/just_bash.ps1:89]: Separate point: auto-upgrading globally installed tooling during setup is a bit invasive. Should we prompt locally or propose an explicit opt-in? Assuming this isn't for CI that is, otherwise I don't understand the point of having this script.

Comment (thms317) [reply template/{{.project_name}}/.just/just_bash.ps1:89]: This case is already covered: after the install/upgrade the script refreshes PATH, runs `just --version` again and throws if the version that actually resolves is still too old. So the "installed new, but PATH still points at old" scenario fails the setup with a clear error instead of passing silently.

Comment (thms317) [reply template/{{.project_name}}/.just/just_bash.ps1:89]: @skinbody I'd keep the auto-upgrade. It only runs when the installed `just` is too old for the template's recipes — the alternative is failing and telling the user to run the exact same winget command by hand. The script also already installs `just` without asking when it's missing, so prompting only for upgrades would be inconsistent. And it runs in CI (that's what the `GITHUB_PATH` handling is for), where a prompt would hang the job. If someone really wants manual control we can add an opt-out env var later.

File: template/{{.project_name}}/.just/just_bash.ps1 (hunk 1 above)
Comment (thms317) [reply template/{{.project_name}}/.just/just_bash.ps1]: Good catch — on a fresh init the file isn't tracked yet, so skip-worktree never applied and the first Windows user's machine-specific bash path would land in the initial commit. There's no good place to hook "after the first commit", so I removed the need for skip-worktree instead: `.just/shell_settings.justfile` is now git-ignored, and `just setup` recreates the default when the file is missing (fresh clones). The whole skip-worktree block is gone.

I also tried the nicer options — a tracked default with a local override import, or `env()`/`os()` in the setting — but `just` rejects both: defining a setting in two imported files is a hard error, and settings can't call functions. So it stays a generated file, just no longer tracked.

File: .justfile (hunk 3 above)
Comment (thms317) [reply .justfile:107]: This was actually a bug: the recipe takes a `profile` argument (`test-deploy profile=PROFILE_NAME:`), but the body used `{{ PROFILE_NAME }}`, so the argument was ignored — `just test-deploy MY_PROFILE` silently used the default. `{{ profile }}` makes the argument work; `PROFILE_NAME` is still the default. And correct, this is the repo-root justfile, separate from the template's `.env` handling — I've added a comment on the recipe saying so.

File: template/{{.project_name}}/.just/just_bash.ps1 (hunk 2 above)
Comment (skinbody) [reply template/{{.project_name}}/.just/just_bash.ps1:89]: @thms317 I understand both points, but changing global installs (installing or updating) without at least a [Y/n] prompt is a big no no. For CI it's fine, though as I said at that point I think there's a bigger question on if this is needed or not

**Review by skinbody — APPROVED** [review]
Looks good, let's rediscuss global tool installs as a separate issue
